xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/Main/Spaces.xml

```diff
@@ -456,10 +456,9 @@ $xwiki.ssx.use('Main.Spaces')##
   ## List spaces.
   ##
   #foreach($space in $spaceList)
-    ## Display space only if the user has view right on space home and if the space is not blacklisted.
-    ## $blacklistedSpaces is set in xwikivars.vm
+    ## Display space only if the user has view right on space home
     #set ($spaceHomeDocumentReference = $services.model.resolveSpace($space))
-    #if($xwiki.hasAccessLevel('view', $services.model.serialize($spaceHomeDocumentReference)) &amp;&amp; !$blacklistedSpaces.contains($space))
+    #if($xwiki.hasAccessLevel('view', $services.model.serialize($spaceHomeDocumentReference)))
```

Comment:
no need to use isHidden here?

Reply:
It's already used directly in the query with a hidden filter, above: https://github.com/xwiki/xwiki-platform/pull/2674/files#diff-1f6ec7d6711ea31424ec12e879a93dccf510c95d9478fcc8e65d90016409387aR452

Note that it's the same in other pages where I just removed that condition: I always checked if there was a filter defined.